src/MongoDB.Driver/MongoClient.cs

```diff
@@ -169,6 +177,14 @@ public sealed override Task DropDatabaseAsync(IClientSessionHandle session, stri
             return ExecuteWriteOperationAsync(session, operation, cancellationToken);
         }
 
+        /// <inheritdoc/>
+        public override ClientEncryption GetClientEncryption(ClientEncryptionOptions clientEncryptionOptions)
+        {
+            var explicitController = new LibMongoCryptController(this, clientEncryptionOptions);
```

Comment:
I was not expecting any new instances of `LibMongoCryptController` to be created. This will result in a new `CryptClient` being created which we are trying to avoid by having `MongoClient` own a single `LibMongoCryptController` which in turn owns a single `CryptClient`.

If the reason we are creating a new `LibMongoCryptController` is because of options perhaps the options need to be removed from the state of `LibMongoCryptController` and passed as parameters to the operations that need the options?